### PR TITLE
Implements DX8Wrapper::Set_Light, corrects float check in Set_Light_Environment

### PR DESCRIPTION
**Pull request by tomsons26**

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
Is this the source of the tree shading bug? Are there any bugs in the issue list this fixes?

File: src/w3d/renderer/dx8wrapper.cpp [hunk 1]
```diff
@@ -1791,10 +1863,10 @@ void DX8Wrapper::Set_Light_Environment(LightEnvironmentClass *light_env)
                 light.Attenuation0 = 1.0f;
 
                 if (fabsf(light_env->Get_Point_Inner_Radius(light_index) - light_env->Get_Point_Outer_Radius(light_index))
-                    >= 0.00001f) {
-                    light.Attenuation1 = 0.1f / light_env->Get_Point_Inner_Radius(light_index);
-                } else {
+                    < 1e-5) {
```
Comment (duncanspumpkin) [inline src/w3d/renderer/dx8wrapper.cpp:1866]: Why invert the logic?

Comment (tomsons26) [reply src/w3d/renderer/dx8wrapper.cpp:1866]: so it follows binary flow

**Comment by tomsons26** [comment]
> Is this the source of the tree shading bug? Are there any bugs in the issue list this fixes?

nah, just cleaning out unmerged things from old branches, this function afict isn't called in game, likely is in w3dviewer i'd guess so yeah needed for later

**Pull request merged**